project1_tiny_calculator/lib/main.dart

```diff
@@ -111,7 +67,7 @@ class _TinyCalculatorState extends State<TinyCalculator> {
                           hintStyle: TextStyle(fontSize: 20),
                           border: InputBorder.none,
                         ),
-                        onChanged: (value) => calculate(),
+                        onChanged: (value) => provider.calculate(),
```

Comment:
you can remove this line

instead, inside the provider itself, you can directly listen on the TextEditingControllers
e.g.
```
firstNumberController.addListener(() { calculate() });
secondNumberController.addListener(() { calculate() });
```

Reply:
did but UI is not responding accordingly